string/string.h

```diff
@@ -0,0 +1,267 @@
+    }
+
+    /**
+     * Retrieves value at specified index
+     * @param index to retrieve element at
+     * @return Element at index or nullptr if invalid
+     */
+    char at(int index) {
+        if (index > len || index < 0) {
+            return NULL_TERMINATOR;
+        }
+
+        return this->string[index];
+    }
+
+    /**
+     * Retrieve a substring of the original string
+     *
+     * @param start index of substring
+     * @param end index of substring
+     * @param substr buffer for storing substring
+     */
+    void substr(int start, int end, char substr[]) {
+        if (len < start || end < start || len < end) return;
+
+        char* substr_head = substr;
+        for (int i = start; i < end; i++) {
+            *substr = string[i];
+            substr++;
+        }
+
+        *substr = NULL_TERMINATOR;
+        substr = substr_head;
+    }
+
+    /**************************************
+     * Element Modifiers
+     **************************************/
+
+    /**
+     * Replaces character at certain position
+     * @param position
+     * @param character
+     */
+    void replace(int position, char character) {
+        this->string[position] = character;
+    }
+
+    /**
+     * Replaces a substring of characters
+     * @param start
+     * @param end
+     * @param character
+     */
+    void replace(int start, int end, char const *character) {
+        if (start < 0 || end > size || end < start) return;
+
+        int i;
+        for (i = start; i < len && i < end; i++) {
+            this->string[i] = *character++;
+        }
+
+        if (i < end) {
+            for (; i < end; i++) {
+                this->string[i] = *character++;
+                this->len++;
+            }
+
+            this->string[this->len] = NULL_TERMINATOR;
+        }
+    }
+
+    /**
+     * Insert a character at a certain index
+     * @param position
+     * @param character
+     */
+    void insert(int position, char character) {
+        if (position > len || len + 1 >= size) return;
+
+        for (int i = len + 2; i > position - 1; i--) {
+            this->string[i] = this->string[i - 1];
+        }
+
+        this->string[position] = character;
+        this->len++;
+    }
+
+    /**
+     * Removes character at position specified
+     * @param position to remove at
+     */
+    void remove(int position) {
+        if (position > len) return;
+
+        for (int i = position; i < len + 1; i++) {
+            this->string[i] = this->string[i + 1];
+        }
+        this->len--;
+    }
+
+    /**
+     * Wipes the entire string buffer
+     */
+    void clear() {
+        int i = 0;
+        while (this->len > 0) {
+            string[i++] = NULL_TERMINATOR;
+            this->len--;
+        }
+    }
+
+    /**
+     * Recount the length of the string in case
+     * null gets randomly added to a string
+     *
+     * @return New Length or -1 if unchanged
+     */
+    int recount() {
```

Comment:
could this just be named "len" or "size"? or does it does it do something different?